data/src/test/java/org/apache/iceberg/io/TestTaskEqualityDeltaWriter.java

```diff
@@ -471,14 +473,31 @@ public void close() throws IOException {
     }
 
     private class GenericEqualityDeltaWriter extends BaseEqualityDeltaWriter {
+
+      private final Schema eqDeleteSchema;
+      private final List<String> eqDeleteColumns;
+
       private GenericEqualityDeltaWriter(PartitionKey partition, Schema schema, Schema eqDeleteSchema) {
         super(partition, schema, eqDeleteSchema);
+        this.eqDeleteSchema = eqDeleteSchema;
+        this.eqDeleteColumns = eqDeleteSchema.columns().stream()
+            .map(Types.NestedField::name).collect(Collectors.toList());
       }
 
       @Override
       protected StructLike asStructLike(Record row) {
         return row;
       }
+
+      @Override
+      protected StructLike asStructLikeKey(Record data) {
```

Comment:
The contract of this method is to return the record as a `StructLike`, not to project data from a row. This implementation cause tests to fail to catch bugs because it is projecting where production code is not expected to project.

Reply:
Are you sure? What I was trying to emulate was the following, though of course we don't have access to `RowDataProjection` or `RowDataWrapper`. This is what `wrap`s the record before it's passed to `deleteKey`.

```java
RowDataProjection.create(schema, deleteSchema);
```

So there's some projection taking place, though this might not be the best place to put it.

Reply:
Yeah, that's why the Flink implementation is this:

```java
return keyWrapper.wrap(data);
```

And the projection is happening in `BaseDeltaTaskWriter`:

```java
if (upsert) {
  writer.deleteKey(deleteSchemaProjection.wrap(row));
}
```

Reply:
Ok. I moved it up to be in the `deleteKey` implementation here.